### PR TITLE
Add basic documentation to classes and utility functions.

### PR DESCRIPTION
**Pull request by TaylorMutch**
This PR combs through a majority of the code base and adds some basic documentation to each class.

### CODE REVIEW

File: source/vistas/core/color.py [hunk 1]
```diff
@@ -117,7 +123,7 @@ def hsva_list(self):
 
 
 def interpolate_color(range, min_color: Color, max_color: Color, t):
-    """ Linearly interpolate in the HSV color space """
+    """ Linearly interpolate in the HSV color space. Copies colors in RGB space to HSV. """
```
Comment (nikmolnar) [inline source/vistas/core/color.py]: "copies" -> "converts"

File: source/vistas/core/math.py [hunk 2]
```diff
@@ -2,12 +2,14 @@
 
 
 def cubic_interpolation(p0: Vector3, p1: Vector3, p2: Vector3, p3: Vector3, t) -> Vector3:
+    """ Obtain the cubic interpolation between 3 vectors as a vector along the parametric parameter t. """
```
Comment (nikmolnar) [inline source/vistas/core/math.py]: "obtain" -> "calculate"

File: source/vistas/core/math.py [hunk 3]
```diff
@@ -2,12 +2,14 @@
...
     t2 = t * t
     a = p3 - p2 - p0 + p1
     return a * t * t2 + (p0 - p1 - a) * t2 + (p2 - p0) * t + p1
 
 
 def catmull_rom_splines(p0: Vector3, p1: Vector3, p2: Vector3, p3: Vector3, t) -> Vector3:
+    """ Obtain the catmull rom spline between 3 vectors as a vector along the parametric parameter t. """
```
Comment (nikmolnar) [inline source/vistas/core/math.py]: "obtain" -> "calculate"

File: source/vistas/core/preferences.py [hunk 4]
```diff
@@ -7,6 +7,8 @@
 
 
 class Preferences:
+    """ Application preferences cache. """
```
Comment (nikmolnar) [inline source/vistas/core/preferences.py]: ```""" Manage key/value preferences """```

File: source/vistas/core/task.py [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 
 class Task:
+    """ Abstract class for handling task management. """
```
Comment (nikmolnar) [inline source/vistas/core/task.py]: It's not abstract.

File: source/vistas/core/timeline.py [hunk 6]
```diff
@@ -7,6 +7,8 @@
 
 
 class Timeline:
+    """ A represenation of a temporal timeline. """
```
Comment (nikmolnar) [inline source/vistas/core/timeline.py]: As opposed to the spatial timeline? The quantum timeline?

Comment (nikmolnar) [inline source/vistas/core/timeline.py]: Comments are good, except when they're not. They should always be run through the filter of "does this explain something not immediately apparent?". If your class is called `Timeline` and your comment is "This is a timeline", the comment is at best unnecessary and at worse a future liability when functionality changes and the comment doesn't.

File: source/vistas/ui/controllers/export.py [hunk 7]
```diff
@@ -16,18 +16,26 @@
 
 
 class ExportTextCtrl(wx.TextCtrl):
+    """ Subclass of wx.TextCtrl with a reference to a text ExportItem. """
```
Comment (nikmolnar) [inline source/vistas/ui/controllers/export.py]: "Subclass of wx.TextCtrl" is immediately apparent from the class definition and should not be duplicated in the docstring.

File: source/vistas/ui/controllers/export.py [hunk 8]
```diff
@@ -16,18 +16,26 @@
...
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.item = None
 
 
 class ExportDeleteTimer(wx.Timer):
+    """ Subclass of wx.Timer with a reference to an ExportTextCtrl. """
```
Comment (nikmolnar) [inline source/vistas/ui/controllers/export.py]: See above.

File: source/vistas/ui/controls/float_ctrl.py [hunk 9]
```diff
@@ -6,6 +6,8 @@
 
 
 class FloatCtrlEvent(FloatCtrlEventBase):
+    """ An event that simply passes a float value. """
```
Comment (nikmolnar) [inline source/vistas/ui/controls/float_ctrl.py]: The value it passes seems less important that what it relates to (really, `value` could be anything). In this case, it's relationship with `FloatCtrl` is more useful information than that floats will be passed to it.

File: source/vistas/ui/controls/gl_camera.py [hunk 10]
```diff
@@ -116,6 +120,10 @@ def Reposition(self):
 
 
 class GLCameraControls(wx.EvtHandler):
+    """
+    Event handler for controlling the the currently enabled CameraInteractor for a given GLCanvas. Handles window
```
Comment (nikmolnar) [inline source/vistas/ui/controls/gl_camera.py]: The fact that this is an event handler (in the wx sense) does not seem like the most pertinent thing to know about this class. I.e., handling events isn't its main function, its how it goes about performing its main function.

File: source/vistas/ui/controls/gl_canvas.py [hunk 11]
```diff
@@ -8,6 +8,10 @@
 
 
 class GLCanvas(wx.glcanvas.GLCanvas):
+    """
+    Main panel for 3D rendering. A shared OpenGL context is created when the first instance of this class is used.
```
Comment (nikmolnar) [inline source/vistas/ui/controls/gl_canvas.py]: It's unclear what "main panel" means here. We use the terms `main_viewer_panel` and `main_panel` elsewhere and this is related to neither.

File: source/vistas/ui/controls/graph_panel.py [hunk 12]
```diff
@@ -6,6 +6,11 @@
 
 
 class GraphPanel(wx.Panel):
+    """
+    Main panel for 2D rendering. GraphPanels render 2D images derived from PIL.Image. Rendering logic is handled by
```
Comment (nikmolnar) [inline source/vistas/ui/controls/graph_panel.py]: See above re: "main panel"

File: source/vistas/ui/utils.py [hunk 13]
```diff
@@ -30,24 +32,30 @@ def make_window_transparent(win):
 
 
 def get_main_window() -> wx.Window:
+    """ A utility function for return the application's MainWindow. """
```
Comment (nikmolnar) [inline source/vistas/ui/utils.py]: Grammer.

File: source/vistas/ui/windows/data_dialog.py [hunk 14]
```diff
@@ -23,6 +24,7 @@ def run(self):
 
 
 class DataDialog(wx.Dialog):
+    """ A Dialog for viewing information about an DataPlugin. """
```
Comment (nikmolnar) [inline source/vistas/ui/windows/data_dialog.py]: "an" -> "a"

File: source/vistas/ui/windows/main.py [hunk 15]
```diff
@@ -17,6 +17,8 @@
 
 
 class MainWindow(wx.Frame):
+    """ The main application window. Handles all customized events that apply across disparate panels and window. """
```
Comment (nikmolnar) [inline source/vistas/ui/windows/main.py]: Not just customized events. Plain 'ol wx events, too.

Comment (nikmolnar) [inline source/vistas/ui/windows/main.py]: And not all of them.

File: source/vistas/core/color.py (hunk 1 above)
Comment (TaylorMutch) [inline source/vistas/core/color.py]: I elected for 'copies' over 'converts' to make it clear that the values based in were not being overwritten. That said, I'm okay with 'converts'.

File: source/vistas/core/timeline.py (hunk 6 above)
Comment (TaylorMutch) [inline source/vistas/core/timeline.py]: Fair enough.

File: source/vistas/core/color.py (hunk 1 above)
Comment (nikmolnar) [inline source/vistas/core/color.py]: The conversion feels like the more salient point. Copying is more an ancillary detail of the conversion mechanism.

Comment (nikmolnar) [inline source/vistas/core/color.py]: Although, I guess that's already implied by "in the HSV color space." In which case maybe the second sentence isn't required at all.

Comment (nikmolnar) [inline source/vistas/core/color.py]: If you feel that conversion implies copy is an important detail to convey, I would do that on the relevant conversion functions themselves.

**Pull request merged**